internal/terminal/table_test.go

```diff
@@ -0,0 +1,264 @@
+        1         
+             [2 3]`,
+		},
+		{
+			description: "Should return a table where the longest length is determined by the longest value in the column as a whole",
+			header:      []string{"short", "longer header that will determine length", "short again"},
+			data: []map[string]interface{}{
+				{"short": 1, "longer header that will determine length": "tiny", "short again": 123},
+				{"short": "the word short is actually longer than the word long", "longer header that will determine length": "tiny again", "short again": 90},
+			},
+			expectedMessage: "\n" + fmt.Sprintf("%s                                                 %s  %s",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("short"),
+					color.New(color.Bold).SprintFunc()("longer header that will determine length"),
+					color.New(color.Bold).SprintFunc()("short again")}...,
+			) + `
+----------------------------------------------------  ----------------------------------------  -----------
+1                                                     tiny                                      123        
+the word short is actually longer than the word long  tiny again                                90         `,
+		},
+		{
+			description: "Should return correctly formatted values in the table",
+			header:      []string{"arrays", "floats", "ints", "maps/objects", "strings"},
+			data: []map[string]interface{}{
+				{
+					"ints":    1,
+					"floats":  12.34,
+					"strings": "tester string",
+					"arrays":  []string{"1", "test", "this"},
+					"maps/objects": struct {
+						test int
+						this float64
+					}{test: 1234, this: 12.345},
+				},
+				{
+					"ints":    000000001,
+					"floats":  12.34,
+					"strings": "test",
+					"arrays":  []int{1, 2, 3, 4},
+					"maps/objects": map[string]int{
+						"test": 1,
+						"this": 2,
+					},
+				},
+				{
+					"ints":    -0002,
+					"floats":  0000123.340000,
+					"strings": "hello",
+					"arrays":  []float64{1, 2.3, 4.5555555555, 6},
+					"maps/objects": map[string][]int{
+						"what happens": {1, 2, 3},
+					},
+				},
+			},
+			expectedMessage: "\n" + fmt.Sprintf("%s                  %s  %s  %s               %s      ",
+				[]interface{}{
+					color.New(color.Bold).SprintFunc()("arrays"),
+					color.New(color.Bold).SprintFunc()("floats"),
+					color.New(color.Bold).SprintFunc()("ints"),
+					color.New(color.Bold).SprintFunc()("maps/objects"),
+					color.New(color.Bold).SprintFunc()("strings"),
+				}...,
+			) + `
+----------------------  ------  ----  -------------------------  -------------
+[1 test this]           12.34   1     {1234 12.345}              tester string
+[1 2 3 4]               12.34   1     map[test:1 this:2]         test         
+[1 2.3 4.5555555555 6]  123.34  -2    map[what happens:[1 2 3]]  hello        `,
+		},
+	} {
+		t.Run(tc.description, func(*testing.T) {
+			table := newTable(tc.header, tc.data)
+			message, err := table.Message()
+			if err != nil {
```

Comment:
the error and success cases should be separate `t.Run`s, instead of using conditional asserts here